solver/src/main/java/org/chocosolver/solver/constraints/set/PropUnionVar.java

```diff
@@ -70,6 +68,20 @@ public PropUnionVar(SetVar union, int vOffset, SetVar indices, int iOffset, SetV
     public void propagate(int evtmask) throws ContradictionException {
         SetVar union = vars[k];
         SetVar indices = vars[k + 1];
+        if (indices.getUB().size() > 0) {
```

Comment:
I think this filtering should be called only once, maybe embedding it in a separate method called depending on the evtmask or an internal variable ?

Reply:
I can do that, indeed